components/arm/universalrobots/ur.go

```diff
@@ -107,6 +107,19 @@ type URArm struct {
 
 const waitBackgroundWorkersDur = 5 * time.Second
 
+// UpdateAction helps hinting the reconfiguration process on what strategy to use given a modified config.
+// See config.UpdateActionType for more information.
+func (ua *URArm) UpdateAction(c *config.Component) config.UpdateActionType {
+	if newCfg, ok := c.ConvertedAttributes.(*AttrConfig); ok {
+		ua.speed = newCfg.Speed
```

Comment:
Note that this will always alter the existing struct even if it turns out that we do not want to keep it. This could cause undesirable behavior.

Consider instead structuring it such that the things that trigger a full reconfigure, do so explicitly, and the struct is not altered unless we definitely want to alter it.

```
		if ua.host != newCfg.Host {
			return config.Reconfigure
		}
		ua.speed = newCfg.Speed
		ua.urHostedKinematics = newCfg.ArmHostedKinematics
		return config.None
```
or

```
		if ua.host == newCfg.Host {
			ua.speed = newCfg.Speed
			ua.urHostedKinematics = newCfg.ArmHostedKinematics
			return config.None
		}
```
I consider the first one slightly more readable and extensible.